scripts/macos_install.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash +x
+
+BINARY_URL="https://github.com/massalabs/thyra/releases/latest/download/thyra-server_darwin"
+SCRIPT="MacOS"
+
+green () { echo -e "\e[01;32m$1:\e[0m $2"; }
+
+fatal () { echo -e "\e[01;31m[$SCRIPT]ERROR:\e[0m $1" >&2; exit 1; }
+
+architecture_version () {
+    case $(uname -m) in
+        x86_64) echo "amd64" ;;
+        arm64)  echo "arm64" ;;
+        *)      fatal "$(uname -m) unsupported." ;;
+    esac
+}
+
+install_thyra () {
+    arch="$(architecture_version)" || exit 1
+    curl -s -L "${BINARY_URL}_${arch}" -o thyra-server || fatal "binary download failed."
+
+    chmod +x thyra-server || fatal "change to executable failed."
+    sudo mv thyra-server /usr/local/bin || fatal "move to /usr/local/bin failed."
+}
+
+configure_start_dnsmasq () {
+    sudo bash -c 'echo "address=/.massa/127.0.0.1" > $(brew --prefix)/etc/dnsmasq.d/massa.conf' || fatal "dnsmas configuration failed."
+    sudo mkdir -p /etc/resolver  || fatal "resolver directory creation failed."
+    sudo bash -c 'echo "nameserver 127.0.0.1" > /etc/resolver/massa'  || fatal "resolver configuration failed."
+
+    sudo brew services start dnsmasq || fatal "dnsmasq service failed to start."
+}
+
+set_local_dns () {
+    case $(sudo lsof -i :53 | sed -n 2p | sed 's/\s.*$//') in
```

Comment:
The sed command is failing on MacOS because `\s` is a GNU extension. So we should use `[[:space:]]` which is supported by POSIX bash, as explained [here](https://stackoverflow.com/a/28256343). 

```suggestion
    case $(sudo lsof -i :53 | sed -n 2p | sed 's/[[:space:]].*$//') in
```